src/vs/workbench/api/common/extHostInlineChat.ts

```diff
@@ -97,12 +97,31 @@ export class ExtHostInteractiveEditor implements ExtHostInlineChatShape {
 		const wrapper = new ProviderWrapper(extension, provider);
 		this._inputProvider.set(wrapper.handle, wrapper);
 		this._proxy.$registerInteractiveEditorProvider(wrapper.handle, metadata?.label ?? extension.displayName ?? extension.name, extension.identifier.value, typeof provider.handleInteractiveEditorResponseFeedback === 'function', typeof provider.provideFollowups === 'function', metadata?.supportReportIssue ?? false);
+		const enablementChangeListener = provider.onDidChangeEnablementStatus(() => {
+			this._proxy.$onDidChangeEnablement(wrapper.handle);
+		});
 		return toDisposable(() => {
 			this._proxy.$unregisterInteractiveEditorProvider(wrapper.handle);
 			this._inputProvider.delete(wrapper.handle);
+			enablementChangeListener.dispose();
 		});
 	}
 
+	async $provideEnablementStatus(handle: number, uri: UriComponents, token: CancellationToken): Promise<true | { reason: string }> {
+		const entry = this._inputProvider.get(handle);
+		if (!entry) {
+			this._logService.warn('CANNOT provide enablement status because the PROVIDER IS GONE');
+			return { reason: 'provider is gone' };
```

Comment:
Use `nls`

Reply:
We don't use NLS for the other logs, should we?

Reply:
Is `reason` only used for logging and does never reached the UI?

Reply:
Oh I see. GitHub shows me like 5 lines here, but you're talking about the `provider is gone` portion. Yes good point